Maskbook/Repository/PersonaRepository.swift

```diff
@@ -296,12 +306,22 @@ enum PersonaRepository {
     
     static func restoreFromJson(_ json: JSON) throws {
         var restoreError: Error?
+        @InjectedProvider(\.personaManager)
+        var personaManager
+        
+        var names = personaManager.personaRecordsSubject.value.compactMap(\.nickname)
+        
         backgroundContext.performAndWait {
             for persona in json.arrayValue {
                 let newPersonaRecord = PersonaRecord(context: backgroundContext)
                 
                 newPersonaRecord.identifier = persona[Persona.CodingKeys.identifier.rawValue].string
-                newPersonaRecord.nickname = persona[Persona.CodingKeys.nickname.rawValue].string
+                
+                let name = persona[Persona.CodingKeys.nickname.rawValue].string
+                let newName = PersonaManager.nonrepeatingName(name: name, fromNames: names)
+                names.append(newName)
```

Comment:
Why append the `newName` to `names`? `names` seems to be untouched after the `newName` generated

Reply:
the old backup files may have duplicated names.
So the rename action should not only compare the local existing names, but also compare the new name generated just now.
`names` change to `usedNames` may be better for understanding.